src/inmanta/server/services/orchestrationservice.py

```diff
@@ -419,6 +561,63 @@ def safe_get(input: JsonType, key: str, default: object) -> object:
 
         return 200
 
+    @handle(methods.put_version, env="tid")
+    async def put_version(
+        self,
+        env: data.Environment,
+        version: int,
+        resources: List[JsonType],
+        resource_state: Dict[ResourceIdStr, const.ResourceState],
+        unknowns: List[Dict[str, PrimitiveTypes]],
+        version_info: JsonType,
+        compiler_version: Optional[str] = None,
+        resource_sets: Optional[Dict[ResourceIdStr, Optional[str]]] = None,
+    ) -> Apireturn:
+        return await self._put_version(
+            env, version, resources, resource_state, unknowns, version_info, compiler_version, resource_sets
```

Comment:
`_put_version` expects a `version_info` argument of type `Optional[model.ModelVersionInfo]`, but this method provides a `JsonType`.

Reply:
so what should I do? I thought I should not change the stable API?

Reply:
Yes. You should not change the signature of a release endpoint in a breaking way or alter it's behavior, but anything else can be done. I think the easiest way forwards here is ass follows:

* Change the type of the `version_info` argument of the `_put_version()` endpoint to `Optional[JsonType]`
* In the handler of the `put_partial()` endpoint: Convert the `version_info` instance to a dictionary (a JsonType) before passing it to the `self._put_version()` method. This can be done using `version_info.dict()`

That way the type annotations are correct. Does that make sense.